src/Stratis.Bitcoin/Connection/ConnectionManager.cs

```diff
@@ -141,6 +141,25 @@ public void Initialize()
 
             this.StartNodeServer();
 
+            // If external IP address supplied this overrides all.
+            if (this.ConnectionSettings.ExternalEndpoint != null)
+            {
+                this.selfEndpointTracker.UpdateAndAssignMyExternalAddress(this.ConnectionSettings.ExternalEndpoint, true);
+            }
+            else
+            {
+                // If external IP address not supplied take first routeable bind address and set score to 10.
+                IPEndPoint nodeServerEndpoint = this.ConnectionSettings.Listen?.FirstOrDefault(x => x.Endpoint.Address.IsRoutable(false))?.Endpoint;
+                if (nodeServerEndpoint != null)
+                {
+                    this.selfEndpointTracker.UpdateAndAssignMyExternalAddress(nodeServerEndpoint, false, 10);
+                }
+                else
+                {
+                    this.selfEndpointTracker.UpdateAndAssignMyExternalAddress(new IPEndPoint(IPAddress.Parse("0.0.0.0").MapToIPv6Ex(), this.Network.DefaultPort), false);
```

Comment:
Now I believe we have here the same mistake as it was in the original code. We must used correct port, not the default one.